### PR TITLE
Generate models

### LINKED ISSUE #1: Data model where we can cause N+1s
We'll need to add migrations and setup a data model where we can cause N+1s easily.

```ruby
# This will N+1 unless we preload author.
article.comments.each do |comment|
  comment.author
end
```

### PR DESCRIPTION
**Pull request by justinkrol**
### Summary
Fixes #1

Use `rails generate model` to generate the models:

```
Users
name:string
email:string

Posts
body:text
user_id:integer
receiver_id:integer

Comments
body: text
user_id:integer
content_id:integer
content_type:string

Likes
user_id:integer
content_id:integer
content_type:string

Photos
post_id:integer
photo_url:string
```

cc @ismail-syed @DerekStride

### CODE REVIEW

File: db/migrate/20170923225655_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :email
```
Comment (DerekStride) [inline db/migrate/20170923225655_create_users.rb]: Can we have one of these be not null?

File: db/migrate/20170923230002_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreatePosts < ActiveRecord::Migration[5.1]
+  def change
+    create_table :posts do |t|
+      t.text :body
+      t.integer :user_id
```
Comment (DerekStride) [inline db/migrate/20170923230002_create_posts.rb]: This should be not null

File: db/migrate/20170923230117_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateComments < ActiveRecord::Migration[5.1]
+  def change
+    create_table :comments do |t|
+      t.text :body
+      t.integer :user_id
+      t.integer :content_id
+      t.string :content_type
```
Comment (DerekStride) [inline db/migrate/20170923230117_create_comments.rb]: These 3 here should be not null

File: db/migrate/20170923230214_create_likes.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateLikes < ActiveRecord::Migration[5.1]
+  def change
+    create_table :likes do |t|
+      t.integer :user_id
+      t.integer :content_id
+      t.string :content_type
```
Comment (DerekStride) [inline db/migrate/20170923230214_create_likes.rb]: not null on these three too

File: db/migrate/20170923230250_create_photos.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreatePhotos < ActiveRecord::Migration[5.1]
+  def change
+    create_table :photos do |t|
+      t.integer :post_id
+      t.string :photo_url
```
Comment (DerekStride) [inline db/migrate/20170923230250_create_photos.rb]: not null on these two too

**Review by DerekStride — CHANGES_REQUESTED** [review]
We should have some of these things be not null

File: db/migrate/20170923225655_create_users.rb (hunk 1 above)
Comment (ismail-syed) [reply db/migrate/20170923225655_create_users.rb]: Hm, good point, I think we should have `firstName` & `lastName`. Where `lastName` can be optional.

So `firstName` & `email` can't be null

**Comment by ismail-syed** [comment]
The not null stuff is validated through model validations, right? Or is this done at the migration level?

eg. `app/models/user.rb`

```
class User < ApplicationRecord
  validates :firstName, presence: true
  validates :email, presence: true
end
```

**Comment by justinkrol** [comment]
^ we talked about this IRL, but yeah good to have it db level as well.

File: db/migrate/20170923230117_create_comments.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class CreateComments < ActiveRecord::Migration[5.1]
+  def change
+    create_table :comments do |t|
+      t.text :body
+      t.integer :user_id, null: false
+      t.integer :content_id, null: false
+      t.string :content_type, null: false
```
Comment (DerekStride) [inline db/migrate/20170923230117_create_comments.rb:7]: Just an FYI, you could have used `t.references` or `t.belongs_to` to achieve the same result for foreign keys.

```ruby
t.references :content, polymorphic: true, null: false
```

**Review by DerekStride — APPROVED** [review]
We never really thought about indexes yet but we can add migrations to add those later.

**Review by DerekStride — CHANGES_REQUESTED** [review]
Actually I just 🎩 'd and the migration needs to be re-run. It changes the `db/schema.rb` file.

**Comment by justinkrol** [comment]
Sorry it took a while to update. Squashed and pushed

**Review by DerekStride — APPROVED** [review]

**Review by ismail-syed — APPROVED** [review]

**Pull request merged**